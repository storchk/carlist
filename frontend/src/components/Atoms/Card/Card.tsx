import { StyledCard, StyledCardText } from './Card.styled'
import type { CardProps } from './Card.types'
export const Card = ({ children, ...props }: CardProps): JSX.Element => {
  return (
    <StyledCard {...props}>
      <div>
        <StyledCardText>{children}</StyledCardText>
      </div>
    </StyledCard>
  )
}
