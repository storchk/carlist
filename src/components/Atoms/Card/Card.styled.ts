import styled, { css } from 'styled-components'

import { StyledTypography } from '../Typography'

export const StyledCardText = styled(StyledTypography)``

export const StyledCard = styled.div(
  ({ theme }) => css`
    display: block;
    transition: all 0.5s cubic-bezier(0.2, 0, 0.05, 1);
    background-color: ${theme.colors.contrast0};
    border-radius: ${theme.border.radius};
    overflow: hidden;
    box-shadow: rgba(0, 0, 0, 0.1) 0px 2px 16px 2px;
    &:hover {
      cursor: pointer;
      box-shadow: rgba(0, 0, 0, 0.1) 0px 0.5rem 8px 0px;
      transform: scale(1.02);
    }
  `
)
