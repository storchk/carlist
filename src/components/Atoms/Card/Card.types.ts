import type { ThemeType } from '../../../theme'

export type CardVariant = 'primary' | 'secondary'

export type CardProps = {
  children: string | JSX.Element | JSX.Element[]
}

