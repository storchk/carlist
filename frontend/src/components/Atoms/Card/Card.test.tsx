import { axe } from 'jest-axe'

import { testing } from '@/testing'

import { Card } from './Card'
import type { CardProps } from './Card.types'

const { render, screen, renderer } = testing
const mockText = 'Lorem Ipsum'

const renderComponent = (props?: Partial<CardProps>) => render(<Card {...props}>{mockText}</Card>)
describe('atoms: Card', () => {
  it('should render button with text', () => {
    renderComponent()
    const text = screen.getByText(/Lorem Ipsum/i)
    expect(text).toBeInTheDocument()
  })

  it('snapshot', () => {
    const view = renderer(<Card>{mockText}</Card>)
    expect(view).toMatchSnapshot()
  })

  it('a11y - No violations detected', async () => {
    const { container } = renderComponent()
    await expect(axe(container)).resolves.toHaveNoViolations()
  })
})
