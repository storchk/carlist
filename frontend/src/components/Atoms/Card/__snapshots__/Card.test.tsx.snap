// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms: Card snapshot 1`] = `
.c1 {
  font-weight: 400;
  color: inherit;
  font-size: inherit;
}

.c0 {
  display: block;
  transition: all 0.5s cubic-bezier(0.2, 0, 0.05, 1);
  background-color: #ffffff;
  border-radius: 8px;
  overflow: hidden;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 2px 16px 2px;
}

.c0:hover {
  cursor: pointer;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0.5rem 8px 0px;
  transform: scale(1.02);
}

<div
  className="c0"
>
  <div>
    <div
      className="c1"
    >
      Lorem Ipsum
    </div>
  </div>
</div>
`;
